maxitg/UsageString
# Full Wolfram Language package

## Changes

* Add package directory hierarchy, which allows one to compile into a paclet.
* Add a documentation page for `UsageString` function.

## Compilation
* Open the project in Wolfram Workbench using File -> Open Projects from File System..., and selecting root repository directory.
* Open Window -> Show View -> Application Tools.
* Select `UsageString` as a project.
* Click Build to build documentation.
* Click Deploy Application, and select a directory to put a temporary deployed package.
* Make sure all files selected, click Next. Make sure documentation is selected, click Finish. A new directory `UsageString` will be created in a directory specified.
* Next, open Mathematica, and evaluate ```pacletPath = PacletManager`PackPaclet["path_to_newly_created_UsageString_directory"]```. The output will be the path to the compiled paclet.
* Next, evaluate ```PacletManager`PacletInstall[pacletPath]``` to install the paclet.

### Tests
* Import the package with ```<<UsageString` ```.
* Evaluate ```UsageString["f[`s`]"]``` to make sure the package code was evaluated.
* Select `UsageString` and press `F1` to open documentation center. The page for `UsageString` should open.
